controller/backup_controller.go

```diff
@@ -701,9 +719,25 @@ func (bc *BackupController) checkMonitor(backup *longhorn.Backup, volume *longho
 	if engine.Status.CurrentState != longhorn.InstanceStateRunning ||
 		engine.Spec.DesireState != longhorn.InstanceStateRunning ||
 		volume.Status.State != longhorn.VolumeStateAttached {
+		backup.Status.State = longhorn.BackupStatePending
+		backup.Status.Messages[MessageTypeReconcileInfo] = fmt.Sprintf(WaitForEngineMessage, engine.Name)
 		return nil, fmt.Errorf("waiting for engine %v to be running before enabling backup monitor", engine.Name)
 	}
 
+	snapshot, err := bc.ds.GetSnapshotRO(backup.Spec.SnapshotName)
+	if err != nil {
+		backup.Status.State = longhorn.BackupStatePending
+		backup.Status.Messages[MessageTypeReconcileInfo] = fmt.Sprintf(SnapshotNotFoundMessage, backup.Spec.SnapshotName)
```

Comment:
`SnapshotNotFoundMessage` is for 'not found' errors. Should we log the appropriate error message for other types of errors?"

Reply:
Fixed, thanks